standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/metrics/CompactionMetricData.java

```diff
@@ -148,6 +152,10 @@ Map<String, Long> getStateCount() {
     return new HashMap<>(stateCount);
   }
 
+  public Map<String, Integer> getPoolCount() {
+    return new HashMap<>(poolCount);
```

Comment:
I wonder, what is the reason of copying the poolCount map instead of passing it to the caller?

Reply:
AS far as I can see `CompactionMetricData` is designed to be immutable, stateCount returned the same way
`return new HashMap<>(stateCount);`

I did not want to break immutability.

Reply:
Good to know, thank you.

Reply:
If the statecount and poolcount are meant to be immutable, why don't we return an immutable hashmap?

Reply:
I think it is enough if only `CompactionMetricData` is immutable. The returned shallow copy allows the caller to modify these collections without modifying CompactionMetricData itself, which is fine in my opinion. However, in the current case both methods are used only one time making this discussion rather theoretical. I simply copied the way the already existing method works.